tests/test_cli.py

```diff
@@ -101,11 +102,56 @@ def test_cli_exit_on_noinput_and_replay(mocker):
         "You can not use both replay and no_input or extra_context "
         "at the same time."
     )
+
     assert expected_error_msg in result.output
 
     mock_cookiecutter.assert_called_once_with(
         template_path,
         None,
         True,
-        replay=True
+        replay=True,
+        overwrite_if_exists=False
     )
+
+@pytest.mark.usefixtures('remove_fake_project_dir')
+def test_cli_exit_on_overwrite_if_exists_and_replay(mocker):
+    mock_cookiecutter = mocker.patch(
+        'cookiecutter.cli.cookiecutter',
+        side_effect=cookiecutter
+    )
+
+    template_path = 'tests/fake-repo-pre/'
+    result = runner.invoke(main, [
+        template_path,
+        '--replay',
+        '-v',
+        '-f',
+    ])
+
+    assert result.exit_code == 0
+
+    mock_cookiecutter.assert_called_once_with(
+        template_path,
+        None,
+        False,
+        replay=True,
+        overwrite_if_exists=True
+    )
+
+
+@pytest.mark.usefixtures('remove_fake_project_dir')
+def test_cli_overwrite_if_exists_when_output_dir_does_not_exist():
+    if os.path.isdir('fake-project'):
```

Comment:
I did not move it into a fixture because there is no other test as of now which uses similar code. Will move it to a fixture if you strongly feel so.

There is already a ``fin_remove_fake_project_dir`` fixture function but it is a nested function and set as a finalizer. So there is no scope to reuse it.